processing/src/main/java/io/druid/segment/data/CompressedIntsIndexedSupplier.java

```diff
@@ -0,0 +1,353 @@
+  }
+
+  public static CompressedIntsIndexedSupplier fromIntBuffer(IntBuffer buffer, final ByteOrder byteOrder, CompressedObjectStrategy.CompressionStrategy compression)
+  {
+    return fromIntBuffer(buffer, MAX_INTS_IN_BUFFER, byteOrder, compression);
+  }
+
+  public static CompressedIntsIndexedSupplier fromIntBuffer(
+      final IntBuffer buffer, final int chunkFactor, final ByteOrder byteOrder, CompressedObjectStrategy.CompressionStrategy compression
+  )
+  {
+    Preconditions.checkArgument(
+        chunkFactor <= MAX_INTS_IN_BUFFER, "Chunks must be <= 64k bytes. chunkFactor was[%s]", chunkFactor
+    );
+
+    return new CompressedIntsIndexedSupplier(
+        buffer.remaining(),
+        chunkFactor,
+        GenericIndexed.fromIterable(
+            new Iterable<ResourceHolder<IntBuffer>>()
+            {
+              @Override
+              public Iterator<ResourceHolder<IntBuffer>> iterator()
+              {
+                return new Iterator<ResourceHolder<IntBuffer>>()
+                {
+                  IntBuffer myBuffer = buffer.asReadOnlyBuffer();
+
+                  @Override
+                  public boolean hasNext()
+                  {
+                    return myBuffer.hasRemaining();
+                  }
+
+                  @Override
+                  public ResourceHolder<IntBuffer> next()
+                  {
+                    IntBuffer retVal = myBuffer.asReadOnlyBuffer();
+
+                    if (chunkFactor < myBuffer.remaining()) {
+                      retVal.limit(retVal.position() + chunkFactor);
+                    }
+                    myBuffer.position(myBuffer.position() + retVal.remaining());
+
+                    return StupidResourceHolder.create(retVal);
+                  }
+
+                  @Override
+                  public void remove()
+                  {
+                    throw new UnsupportedOperationException();
+                  }
+                };
+              }
+            },
+            CompressedIntBufferObjectStrategy.getBufferForOrder(byteOrder, compression, chunkFactor)
+        ),
+        compression
+    );
+  }
+
+  public static CompressedIntsIndexedSupplier fromList(
+      final List<Integer> list , final int chunkFactor, final ByteOrder byteOrder, CompressedObjectStrategy.CompressionStrategy compression
+  )
+  {
+    Preconditions.checkArgument(
+        chunkFactor <= MAX_INTS_IN_BUFFER, "Chunks must be <= 64k bytes. chunkFactor was[%s]", chunkFactor
+    );
+
+    return new CompressedIntsIndexedSupplier(
+        list.size(),
+        chunkFactor,
+        GenericIndexed.fromIterable(
+            new Iterable<ResourceHolder<IntBuffer>>()
+            {
+              @Override
+              public Iterator<ResourceHolder<IntBuffer>> iterator()
+              {
+                return new Iterator<ResourceHolder<IntBuffer>>()
+                {
+                  int position = 0;
```

Comment:
no, this has not changed compared to other columns. The role of the supplier is to provide one instance per thread.